db/sql/init_db.sql

```diff
@@ -37,4 +37,59 @@ CREATE TABLE IF NOT EXISTS credentials (
     created_at TIMESTAMP DEFAULT NOW(),
     updated_at TIMESTAMP DEFAULT NOW(),
     deleted_at TIMESTAMP
-);
\ No newline at end of file
+);
+
+-- COURSES TABLE
+CREATE TABLE IF NOT EXISTS courses (
+    id SERIAL PRIMARY KEY,
+    course_code VARCHAR NOT NULL,          
+    course_name VARCHAR NOT NULL,          
+    term VARCHAR NOT NULL,                 
+    section VARCHAR,                       
+    instructor_id INTEGER NOT NULL REFERENCES users(id) ON DELETE CASCADE,
+    start_date DATE,
+    end_date DATE,
+    created_at TIMESTAMP DEFAULT NOW(),
+    updated_at TIMESTAMP DEFAULT NOW(),
+    deleted_at TIMESTAMP,
+    UNIQUE (course_code, term, section)
+);
+
+-- TEAMS TABLE
+CREATE TABLE IF NOT EXISTS teams (
+    id SERIAL PRIMARY KEY,
+    course_id INTEGER NOT NULL REFERENCES courses(id) ON DELETE CASCADE,
+    name VARCHAR NOT NULL,                 
+    description TEXT,
+    created_at TIMESTAMP DEFAULT NOW(),
+    updated_at TIMESTAMP DEFAULT NOW(),
+    deleted_at TIMESTAMP,
+    UNIQUE (course_id, name)
+);
+
+-- ENROLLMENTS TABLE
+-- One row per user per course (optionally tied to a team)
+CREATE TABLE IF NOT EXISTS enrollments (
+    id SERIAL PRIMARY KEY,
+    user_id INTEGER NOT NULL REFERENCES users(id) ON DELETE CASCADE,
+    course_id INTEGER NOT NULL REFERENCES courses(id) ON DELETE CASCADE,
+    team_id INTEGER REFERENCES teams(id) ON DELETE SET NULL,
+    role VARCHAR NOT NULL DEFAULT 'student',   
+    enrolled_at TIMESTAMP DEFAULT NOW(),
```

Comment:
This would just be the same as created_at, right? I don't think we need this column then.